### ISSUE TITLE
Codejoc Controls 20.3.0

### ISSUE THREAD

**Issue by verticalsoft** [twinbasic/twinbasic#1416]
Hello I have an issue with the codejock commandbars control in that under VB6 I can invoke some addiation methods which are unavailbale currently under TB.

I've attached a picture to better describe the issue.

On some of the other controls you get the option for ActiveX Properties...

![VB6](https://user-images.githubusercontent.com/5321212/207006585-2d49afcd-285d-4414-b8c7-6458852442bc.jpg)
![TB](https://user-images.githubusercontent.com/5321212/207006734-fc7e1fea-b3cf-4ad9-97c9-6eb2dbbd9261.jpg)

**Comment by fafalone** [comment]
In VB6 go to the Add-Ins menu, click Add In Manager, and see if there's anything from Codejock... because that seems like an addin rather than something the controls themselves could do.  (.dsr files- the main source file for addins- are 'Designer modules')

I don't think there's plans to support these in tB since it's an entirely different IDE.

**Comment by wqweto** [comment]
I think that that the container (the IDE) can use [IOleObject::EnumVerbs](https://learn.microsoft.com/en-us/windows/win32/api/oleidl/nf-oleidl-ioleobject-enumverbs) method to integrate custom menu items in the control context menu besides system provided `OLEIVERB_PROPERTIES`.

Here I have DataDynamics ActiveBar control which does add a custom menu item (`Designer`) on context popup without having a helper add-in installed.

**Comment by fafalone** [comment]
Interesting... never seen such a thing before. Are there any code examples of it? Initial search didn't turn up any. Presumably you'd need to hook IOleObject to supply a response to EnumVerbs?

**Comment by verticalsoft** [comment]
You can download evaluation versions of Codejock ActiveX and see the menus in VB6.
The options I've posted on the opening thread are all available proramatically apart from Edit, when you select edit you are taken to a command bar editor and once finished and closed the objectes created in the editor are stored in the activex control on the form.

You can get over the edit function as codejock do actually provide a CommandBar Editor which can save the files and then import them programatically.

How you go about doing this in TB I'll leave to the experts, I just stand on the sholders of giants

![image](https://user-images.githubusercontent.com/5321212/207363709-d9707d89-b1ac-44e3-b279-9913f8e5e1f5.png)

**Comment by wqweto** [comment]
> Since UCs have internal implementations of IOleObject presumably you'd need to hook it and to supply a response pointing to an IEnumOLEVERB object you've created, and DoVerb to respond?

Yes, nothing short of hooking the interface can customize this for VB6 user-controls.

Btw, I'm not 100% positive this is the mechanism VB6 IDE exposes custom menu items but seems plausible. Still might be a dead end.
